routes.py

```diff
@@ -0,0 +1,663 @@
+    profiles = Profile.query.filter_by(user_id=user.id).all()
+    if len(profiles) == 0:
+        profiles = []
+    form = ProfileForm()
+    return render_template('user.html', profiles=profiles, form=form, user=user, get_server_short_name=get_server_short_name)
+
+
+# ścieżka pozwalająca tworzyć profile i przekierowywuje spowrotem na stronę użytkownika
+@app.route('/user/<username>/create_profile', methods=['GET', 'POST'])
+@login_required
+def create_profile(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    form = ProfileForm()
+    if form.validate_on_submit():
+        existing_profile = Profile.query.filter_by(
+            nickname=form.nickname.data, server=form.server.data).first()
+        if existing_profile is not None:
+            flash('A profile with this nickname already exists on this server.', 'error')
+        else:
+            new_profile = Profile(
+                nickname=form.nickname.data,
+                server=form.server.data,
+                division=form.division.data,
+                best_position=form.pref_role.data,
+                alternative_position=form.alternative_role.data,
+                user_id=user.id
+            )
+            db.session.add(new_profile)
+            db.session.commit()
+            flash('Your profile has been created successfully.', 'success')
+    return redirect(url_for('user', username=user.username))
+
+
+# ścieżka edycji profilu
+@app.route('/user/<username>/<int:profile_id>/edit_profile', methods=['GET', 'POST'])
+@login_required
+def edit_profile(username, profile_id):
+    user = User.query.filter_by(username=username).first_or_404()
+    profile = Profile.query.filter_by(id=profile_id).first_or_404()
+    form1 = EditProfileForm()
+    form2 = DeleteProfileForm()
+    if form1.validate_on_submit():
+        existing_profile = Profile.query.filter_by(
+            nickname=form1.nickname.data, server=form1.server.data).first()
+        if existing_profile is not None:
+            flash('A profile with this nickname already exists on this server.', 'error')
+        else:
+            profile.nickname = form1.nickname.data
+            profile.server = form1.server.data
+            profile.division = form1.division.data
+            profile.best_position = form1.pref_role.data
+            profile.alternative_position = form1.alternative_role.data
+
+            db.session.add(profile)
+            db.session.commit()
+            flash('Your profile has been updated.', 'success')
+        return redirect(url_for('user', username=user.username))
+    elif form2.delete_button.data:
+        db.session.delete(profile)
+        db.session.commit()
+        flash('Your profile has been deleted.', 'message')
+        return redirect(url_for('user', username=user.username))
+    return render_template('profile_edit.html', profile=profile, form1=form1, form2=form2, user=user)
```

Comment:
Try to rewrite the function above (and others, where appropriate) using [early returns](http://www.itamarweiss.com/personal/2018/02/28/return-early-pattern.html). It makes code a lot easier to read.